Nessie server attempts to initialize Mongo Client even when not requested

Running the following command is supposed to start Nessie Server with in-memory store (default). However, it attempts to connect to MongoDB nonetheless.

```
$ java -jar servers/quarkus-server/target/quarkus-app/quarkus-run.jar
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2021-10-20 21:09:04,203 INFO  [org.mon.dri.cluster] (main) Cluster created with settings {hosts=[127.0.0.1:27017], mode=SINGLE, requiredClusterType=UNKNOWN, serverSelectionTimeout='30000 ms'}
2021-10-20 21:09:05,653 INFO  [org.mon.dri.cluster] (cluster-ClusterId{value='6170bdb046cda22777383ce6', description='null'}-127.0.0.1:27017) Exception in monitor thread while connecting to server 127.0.0.1:27017: com.mongodb.MongoSocketOpenException: Exception opening socket
	at com.mongodb.internal.connection.SocketStream.open(SocketStream.java:70)
	at com.mongodb.internal.connection.InternalStreamConnection.open(InternalStreamConnection.java:165)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.lookupServerDescription(DefaultServerMonitor.java:195)
	at com.mongodb.internal.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:151)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
	at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:399)
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:242)
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:224)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.base/java.net.Socket.connect(Socket.java:609)
	at com.mongodb.internal.connection.SocketStreamHelper.initialize(SocketStreamHelper.java:107)
	at com.mongodb.internal.connection.SocketStream.initializeSocket(SocketStream.java:79)
	at com.mongodb.internal.connection.SocketStream.open(SocketStream.java:65)
	... 4 more
```

--- a/servers/quarkus-server/src/main/java/org/projectnessie/server/providers/QuarkusMongoDatabaseClient.java
+++ b/servers/quarkus-server/src/main/java/org/projectnessie/server/providers/QuarkusMongoDatabaseClient.java
@@ -16,6 +16,9 @@
 package org.projectnessie.server.providers;
 
 import com.mongodb.client.MongoClient;
+import io.quarkus.arc.Arc;
+import io.quarkus.mongodb.runtime.MongoClientBeanUtil;
+import io.quarkus.mongodb.runtime.MongoClients;
 import javax.inject.Inject;
 import javax.inject.Singleton;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
@@ -27,8 +30,11 @@ import org.projectnessie.versioned.persist.mongodb.MongoDatabaseClient;
 public class QuarkusMongoDatabaseClient extends MongoDatabaseClient {
   @Inject
   public QuarkusMongoDatabaseClient(
-      @ConfigProperty(name = "quarkus.mongodb.database") String databaseName,
-      MongoClient mongoClient) {
+      @ConfigProperty(name = "quarkus.mongodb.database") String databaseName) {
+    MongoClients mongoClients = Arc.container().instance(MongoClients.class).get();
+    MongoClient mongoClient =
+        mongoClients.createMongoClient(MongoClientBeanUtil.DEFAULT_MONGOCLIENT_NAME);
+
     configure(MongoClientConfig.of(mongoClient).withDatabaseName(databaseName));
     initialize();
   }

